crates/iota-storage/src/http_key_value_store.rs

```diff
@@ -86,86 +86,184 @@ pub enum Key {
+    ///     encoded_tagged_key(&TaggedKey::CheckpointSequenceNumber(chk_seq_num));
+    /// let (resource_type, encoded_key_digest) = key.to_path_elements();
+    /// assert_eq!(resource_type, "cs");
+    /// assert_eq!(encoded_key_digest, expected_encoded_seq_num);
+    /// ```
+    pub fn to_path_elements(&self) -> (&'static str, String) {
+        let encoded_key_digest = match self {
+            Key::Tx(digest) => encode_digest(digest),
+            Key::Fx(digest) => encode_digest(digest),
+            Key::CheckpointContents(seq) => {
+                encoded_tagged_key(&TaggedKey::CheckpointSequenceNumber(*seq))
+            }
+            Key::CheckpointSummary(seq) => {
+                encoded_tagged_key(&TaggedKey::CheckpointSequenceNumber(*seq))
+            }
+            Key::CheckpointSummaryByDigest(digest) => encode_digest(digest),
+            Key::TxToCheckpoint(digest) => encode_digest(digest),
+            Key::ObjectKey(object_id, version) => encode_object_key(object_id, version),
+            Key::EventsByTxDigest(digest) => encode_digest(digest),
+        };
 
-fn key_to_path_elements(key: &Key) -> IotaResult<(String, &'static str)> {
-    match key {
-        Key::Tx(digest) => Ok((encode_digest(digest), "tx")),
-        Key::Fx(digest) => Ok((encode_digest(digest), "fx")),
-        Key::CheckpointContents(seq) => Ok((
-            encoded_tagged_key(&TaggedKey::CheckpointSequenceNumber(*seq)),
-            "cc",
-        )),
-        Key::CheckpointSummary(seq) => Ok((
-            encoded_tagged_key(&TaggedKey::CheckpointSequenceNumber(*seq)),
-            "cs",
-        )),
-        Key::CheckpointSummaryByDigest(digest) => Ok((encode_digest(digest), "cs")),
-        Key::TxToCheckpoint(digest) => Ok((encode_digest(digest), "tx2c")),
-        Key::ObjectKey(object_id, version) => Ok((encode_object_key(object_id, version), "ob")),
-        Key::EventsByTxDigest(digest) => Ok((encode_digest(digest), "evtx")),
+        (self.item_type(), encoded_key_digest)
     }
 }
 
-// Create a a [`Key`] instance based on the provided [`base64_url`] encoded
-/// string and item type.
-pub fn path_elements_to_key(digest: &str, item_type: &str) -> anyhow::Result<Key> {
-    let decoded_key = base64_url::decode(digest)
-        .map_err(|err| anyhow::anyhow!("invalid base64 url string: {err}"))?;
-
-    match item_type {
-        "tx" => Ok(Key::Tx(TransactionDigest::try_from(
-            decoded_key.as_slice(),
-        )?)),
-        "fx" => Ok(Key::Fx(TransactionDigest::try_from(
-            decoded_key.as_slice(),
-        )?)),
-        "cc" => {
-            let tagged_key = bcs::from_bytes(&decoded_key).map_err(|err| {
-                anyhow::anyhow!("failed to deserialize checkpoint sequence number: {err}")
-            })?;
-            match tagged_key {
-                TaggedKey::CheckpointSequenceNumber(seq) => Ok(Key::CheckpointContents(seq)),
+impl TryFrom<(&str, &str)> for Key {
```

Comment:
tbh I'd rather have a method called `try_from` with two params than this impl as it is extremely unclear to the caller

Reply:
@DaughterOfMars 

[fixed](https://github.com/iotaledger/iota/pull/5296/commits/d6aee8a5ea73f6460cf684cda12b9be1487a46dd)